modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/view_file_entry.jsp

```diff
@@ -48,7 +51,7 @@ if (portletTitleBasedNavigation) {
 
 	<portlet:actionURL name="/document_library/edit_file_entry" var="editFileEntry" />
 
-	<aui:form action="<%= editFileEntry %>" method="post" name="fm">
+	<aui:form action="<%= editFileEntry %>" method="post" name="fm" portletNamespace="<%= portletNamespace + randomNamespace %>">
```

Comment:
the random portletNamespace ensures that the namespaces of the form's fields are unique. Without them, warning messages like these will appear on the browser console:
```
Found 2 elements with non-unique id #_com_liferay_asset_publisher_web_portlet_AssetPublisherPortlet_INSTANCE_qrsa_cmd
Found 2 elements with non-unique id #_com_liferay_asset_publisher_web_portlet_AssetPublisherPortlet_INSTANCE_qrsa_fileEntryId
Found 2 elements with non-unique id #_com_liferay_asset_publisher_web_portlet_AssetPublisherPortlet_INSTANCE_qrsa_fm
```

Reply:
Splitting hairs a bit, but the word "ensures" isn't _quite_ accurate (I can roll a dice twice and get the same random number "3" two times in a row, after all 😉)

Obviously, not likely to be a problem here [because the implementation of `randomId()`](https://github.com/liferay/liferay-portal/blob/2135e256e6c586c2fa72d2a03344dfd021ffb1f6/portal-kernel/src/com/liferay/portal/kernel/util/StringUtil.java#L2120-L2141) would require us to get the same sequence of 4 letters on two occasions...

![just-sayin](https://user-images.githubusercontent.com/7074/105989794-e69d4600-60a1-11eb-934c-14f1a16b0463.gif)

Reply:
@IstvanD My concern here is that the form action will no longer work correctly. It works in `comment-taglib` because we are [passing](https://github.com/liferay/liferay-portal/blob/master/modules/apps/comment/comment-taglib/src/main/resources/META-INF/resources/discussion/page.jsp#L64) `randomNamespace` to backend and explicitely handling it there.

I don't think this form will even get to backend. It is used in the `move` action in JS. In it, we are [namespacing](https://github.com/liferay-frontend/liferay-portal/blob/011b66f817cf63c8a72d8a37227d866818db0498/modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/view_file_entry.jsp#L176) with `<portlet:namespace />` and not random namespace, so `document.getElementById` will not find the form.

Reply:
I see. Should then the randomNamespace be added to the name of the form's [individual fields](https://github.com/liferay-frontend/liferay-portal/blob/011b66f817cf63c8a72d8a37227d866818db0498/modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/view_file_entry.jsp#L55-L61)?
The browser will warn of the duplication for each 7 of them.

Reply:
I don't think there is an elegant solution. A convention we normally follow is to have only one form per page, this is one of the reasons why.

What I would do is remove the form entirely, replacing it with `Liferay.Util.fetch`, like [here](https://github.com/liferay/liferay-portal/blob/master/modules/apps/journal/journal-content-web/src/main/resources/META-INF/resources/journal_template.jsp#L121-L140). Of course I am assuming here that the form is used only in the `move` action. This is probably beyond the scope of this ticket, could be done in a separate one.

Reply:
Thank you for the explanation. I'll then revert this change as well.